pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -1124,7 +1185,13 @@ func (cloud *CloudProvider) CreateImage(ctx context.Context, project string, vol
 		return nil, err
 	}
 
-	return cloud.waitForImageCreation(ctx, project, imageName)
+	newImage, err := cloud.waitForImageCreation(ctx, project, imageName)
+
+	if err == nil {
+		err = cloud.attachTagsToResource(ctx, snapshotParams.ResourceTags, project, newImage.Id, imagesType, "", false, resourceManagerHostSubPath)
```

Comment:
What error will be reported if the attachTagsToResource call fails? We should ensure that this only returns retryable errors (eg: non-final), so that snapshot/image creation can be re-attempted.

See https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/1558 for an example where this was updated for CreateVolume